From 8630ed0993437f5cd3b15aa7f3363c4421d8519a Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Mon, 8 Jan 2024 06:53:37 +0200
Subject: [PATCH 26/26] savecompat: Add wlcb for all cities

It's a mandatory field, so failure to add it for non-conquered
cities lead to savegame loading failure.

See RM #143

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 server/savegame/savecompat.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/server/savegame/savecompat.c b/server/savegame/savecompat.c
index 26b1d1e7cb..93bbab9f8e 100644
--- a/server/savegame/savecompat.c
+++ b/server/savegame/savecompat.c
@@ -2375,14 +2375,15 @@ static void compat_load_030200(struct loaddata *loading,
             secfile_insert_int(loading->file, CACQ_CONQUEST,
                                "player%d.c%d.acquire_t",
                                plrno, cnro);
-            secfile_insert_int(loading->file, WLCB_SMART,
-                               "player%d.c%d.wlcb",
-                               plrno, cnro);
           } else {
             secfile_insert_int(loading->file, CACQ_FOUNDED,
                                "player%d.c%d.acquire_t",
                                plrno, cnro);
           }
+
+          secfile_insert_int(loading->file, WLCB_SMART,
+                             "player%d.c%d.wlcb",
+                             plrno, cnro);
         }
       }
 
-- 
2.43.0

